src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -57,6 +57,12 @@ public override void Dispose()
                     _sslStream.Dispose();
                     _sslStream = null;
                 }
+
+                if (_tcpStream != null)
+                {
+                    _tcpStream.Dispose();
```

Comment:
Yes, but _stream isn't nulled out, which means code could still end up referencing it, even though both _tcpStream and _sslStream have been disposed and nulled out.  Is that ok?